UndertaleModLib/Models/UndertaleShader.cs

```diff
@@ -528,10 +528,21 @@ public void WriteData(UndertaleWriter writer)
             writer.Write(Data);
         }
 
-        public void ReadData(UndertaleReader reader, int length)
+        public void ReadData(UndertaleReader reader, int length, bool isLast = false)
         {
-            if (_Length != 0 && _Length != length)
-                throw new UndertaleSerializationException("Failed to compute length of shader data");
+            if (_Length != 0) // _Length is the "expected" length here
+            {
+                if (_Length > length)
+                    throw new UndertaleSerializationException("Failed to compute length of shader data: instructed to read less data than expected.");
+                else if (_Length < length)
+                {
+                    if (isLast && ((reader.AbsPosition + length) % 16 == 0)) // Normal for the last element due to chunk padding, just trust the system
+                        length = (int)_Length;
+                    else
+                        throw new UndertaleSerializationException("Failed to compute length of shader data: instructed to read more data than expected." +
+                            (isLast ? " Shader data was the last in the shader, but given length was incorrectly padded." : ""));
+                }
```

Comment:
just noticed: can you make this slightly easier to read by using curly brackets please?